backend/src/Designer/Controllers/RepositoryController.cs

```diff
@@ -558,8 +559,11 @@ public ActionResult ContentsZip(string org, string repository, [FromQuery] bool
                 return BadRequest("User does not have a local clone of the repository.");
             }
 
-            var outStream = new MemoryStream();
-            using (var archive = new ZipArchive(outStream, ZipArchiveMode.Create, leaveOpen: true))
+            var zipType = full ? "full" : "changes";
+            var zipFileName = $"{org}-{repository}-{zipType}.zip";
+            var zipFilePath = Path.Combine(Path.GetTempPath(), zipFileName);
+
+            using (var archive = new ZipArchive(new FileStream(zipFilePath, FileMode.Create, FileAccess.ReadWrite, FileShare.Read), ZipArchiveMode.Create, leaveOpen: false))
```

Comment:
Shouldn't this filestream be deleted after returning from response? It should be created with DeleteOnClose option

Reply:
fixed